grin/src/server.rs

```diff
@@ -132,15 +136,16 @@ impl Server {
 				);
 			}
 			Seeding::WebStatic => {
-				seed.connect_and_monitor(evt_handle.clone(), seed::web_seeds(evt_handle.clone()));
+				seed.connect_and_monitor(
+					evt_handle.clone(),
+					seed::web_seeds(evt_handle.clone()),
+				);
 			}
 			_ => {}
 		}
 
-		if config.seeding_type != Seeding::None {
-			let sync = sync::Syncer::new(shared_chain.clone(), p2p_server.clone());
-			net_adapter.start_sync(sync);
-		}
+		let sync = sync::Syncer::new(shared_chain.clone(), p2p_server.clone());
+		net_adapter.start_sync(sync);
```

Comment:
I understand you might still want to monitor connection but why seeding in this case?

Reply:
I'm not totally sure I understand the question - but are you asking about why I'm proposing to run the sync _from_ a node with no seeds?

This is what i was seeing in testing - 

3 nodes -
* S (seed)
* M (miner)
* N (regular node)

Let them all run for an hour -
* S: height 60
* M: height 60
* N: height 60

Now stop the seed and let the network run for a few minutes -
* S: height 60
* M: height 65
* N: height 65

Restarting the seed (current behavior, no sync running on S) -
* S: height 60
* M: height 65
* N: height 65

Then stop the regular node N, let the network run for a few mins -
* S: height 60
* M: height 70
* N: height 65

When N restarts it successfully peers against S and M and will request blocks from both as it attempts to resync.

S is out of sync and will never resync.
N re-syncs slowly as it is is unaware that S is not a useful peer to sync from.

Reply:
Makes sense, thanks for clarifying. However this will still be a problem for a single mining node, as sync will try to get peers forever and block the start of the miner loop.

Reply:
Ahh ok - I see your concern. Let me take another look at this.

Reply:
One suggestion: instead of removing that condition, just add a check on the number of peers in db and only skip sync if we have no seeds and no known peers.